scripts/FigS2B_adt-plots.R

```diff
@@ -0,0 +1,144 @@
+    scpca_filter == "Remove" & adt_scpca_filter == "Keep" ~ filter_levels[3],
+    scpca_filter == "Keep" & adt_scpca_filter == "Remove" ~ filter_levels[4],
+    TRUE ~ "Error"
+  ),
+  # make sure Keep is first 
+  filter_summary = forcats::fct_relevel(filter_summary, "Keep"),
+  keep_column = ifelse(
+    stringr::str_starts(filter_summary, "Keep"),
+    "Keep",
+    "Remove"
+  ))
+
+# faceted plot to show cells belonging to each filter group
+filtered_plot <- ggplot(filtered_coldata_df, 
+                        aes(x = detected, y = subsets_mito_percent, color = keep_column)) +
+  geom_point(alpha = 0.5, size = 1) +
+  labs(
+    x = "Number of genes detected",
+    y = "Mitochondrial percentage"
+  ) +
+  facet_wrap(vars(filter_summary)) +
+  theme(
+    plot.title = element_text(hjust = 0.5),
+    legend.position = "none"
+  )
+
+# Top ADTs ---------------------------------------------------------------------
+
+# Calculate variance for each ADT
+adt_var <- altExp(processed_sce) |>
+  logcounts() |>
+  apply(1, var, na.rm = TRUE)
+
+# Get the top 4
+top_adts <- adt_var[order(adt_var, decreasing = TRUE)[1:4]] |>
+  names()
+
+# ADT expression density -------------------------------------------------------
+
+# grab expression for top ADTs from counts
+var_adt_exp_df <- logcounts(altExp(processed_sce))[top_adts, ] |>
+  # as.matrix needs to come _first_
+  as.matrix() |>
+  t() |>
+  as.data.frame() |>
+  tibble::rownames_to_column("barcodes") |>
+  # combine all ADTs into a single column for  faceting
+  tidyr::pivot_longer(
+    !barcodes,
+    names_to = "ADT",
+    values_to = "adt_expression"
+  )
+
+adt_density_plot <- ggplot(var_adt_exp_df, aes(x = adt_expression, fill = ADT)) +
+  geom_density() +
+  facet_wrap(vars(ADT), nrow = 2) +
+  labs(x = "Log-normalized ADT expression",
+       y = "Density") +
+  theme(legend.position = "none")
+
+# ADT expression UMAP ----------------------------------------------------------
+
+# create data frame of UMAPs and expression
+umap_df <- scuttle::makePerCellDF(processed_sce) |>
+  dplyr::select(
+    barcodes,
+    UMAP1 = UMAP.1,
+    UMAP2 = UMAP.2
+  ) |>
+  # combine with gene expression
+  dplyr::left_join(var_adt_exp_df, by = "barcodes")
+
+adt_umap_plot <- ggplot(umap_df, aes(x = UMAP1, y = UMAP2, color = adt_expression)) +
+  geom_point(alpha = 0.1, size = 0.1) +
+  facet_wrap(vars(ADT)) +
+  scale_color_viridis_c() +
+  labs(
+    color = "Log-normalized ADT expression"
+  ) +
+  # remove axis numbers and background grid
+  scale_x_continuous(labels = NULL, breaks = NULL) +
+  scale_y_continuous(labels = NULL, breaks = NULL) +
+  coord_fixed() +
+  theme(
+    legend.position = "none"
+  )
+
+# Combine and export -----------------------------------------------------------
+
+combined_plot <- patchwork::wrap_plots(list(filtered_plot, adt_density_plot, adt_umap_plot))
+
+ggsave(output_plot_file, combined_plot)
```

Comment:
When I run the script, the text doesn't fit - and it's because my exported figure size is different from yours. We'll need height & width arguments here to ensure everything fits, as well as associated reproducibility 

![FigS2B_adt-plots](https://github.com/AlexsLemonade/scpca-paper-figures/assets/4701111/ae52fb8f-c7e6-447a-a2fe-88fb99228eb7)